internal/git/gitsync_processor.go

```diff
@@ -369,11 +383,27 @@ func CheckRepoUpdatesForKustomize(ctx context.Context, r *git.Repository, specs
 // this check for file changes in the repo by comparing the old commit hash with new commit hash
 //and returns the recent hash with PatchedResources map
 
-func CheckForRepoUpdates(ctx context.Context, r *git.Repository, specs *v1alpha1.GitSyncSpec, lastCommitHash string, defaultNameSpace string) (PatchedResource, string, error) {
+func CheckForRepoUpdates(ctx context.Context, r *git.Repository, specs *v1alpha1.GitSyncSpec, lastCommitHash string, defaultNameSpace string, kubeClient kubernetes.Client) (PatchedResource, string, error) {
 	var patchedResources PatchedResource
 	var recentHash string
 	logger := logging.FromContext(ctx).With("Repository Url", specs.RepoUrl)
-	if err := fetchUpdates(r); err != nil {
+
+	// Get the latest config from global config manager
+	globalConfig, err := controllerconfig.GetConfigManagerInstance().GetConfig()
+	if err != nil {
+		logger.Errorw("error getting the global Config", "err", err)
+		return PatchedResource{}, "", fmt.Errorf("error getting the global Config %s", err)
+	}
+
+	credentials := gitshared.FindCredByUrl(specs.RepoUrl, globalConfig)
+
+	fetchOptions, err := gitshared.GetRepoFetchOptions(ctx, credentials, kubeClient, defaultNameSpace, specs.RepoUrl)
+	if err != nil {
+		logger.Errorw("error fetching the updates from  the github repo", "err", err)
+		return PatchedResource{}, "", fmt.Errorf("error getting fetch options %s", err)
+	}
```

Comment:
Nit: would it be possible to consolidate the duplicate code from here and `CheckRepoUpdatesForKustomize()` into a single method?

Reply:
Maybe we can just add the call for getting the current ConfigMap values to `GetRepoFetchOptions()`? What do you think?